frontend/src/api/k8s/inferenceServices.ts

```diff
@@ -45,6 +45,13 @@ export const assembleInferenceService = (
   const dataConnectionKey = secretKey || dataConnection;
 
   const nonEmptyArgs = servingRuntimeArgs?.filter(Boolean) || [];
+  // If we have nonEmptyArgs, ensure that we properly handle separating args
+  let splitArgs: string[] = [];
+  if (nonEmptyArgs.length > 0) {
+    for (let i = 0; i < nonEmptyArgs.length; i++) {
+      splitArgs = splitArgs.concat(parseCommandLine(nonEmptyArgs[i]));
+    }
+  }
```

Comment:
We can simplify this using [flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap).

```ts
const splitArgs = nonEmptyArgs.flatMap(parseCommandLine);
```